False positive on sorted_first_last with first(where:)

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `sorted_first_last` shouldn't trigger if there is a predicate passed to the `first` call

##### Complete output when running SwiftLint, including the stack trace and command used

**`MyPlayground.swift`**
```swift
import Foundation

struct RaceResult {
    let name: String
    let finish: Date
    let disqualified: Bool
}

let logbook = [
    RaceResult(name: "Troy", finish: Date.init(timeIntervalSince1970: 300), disqualified: false),
    RaceResult(name: "Rachel", finish: Date.init(timeIntervalSince1970: 100), disqualified: true),
    RaceResult(name: "Jessie", finish: Date.init(timeIntervalSince1970: 200), disqualified: false),
    RaceResult(name: "Tiffany", finish: Date.init(timeIntervalSince1970: 400), disqualified: false)
]

let winner = logbook
    .sorted { lhs, rhs -> Bool in
        lhs.finish < rhs.finish
    }
    .first { result -> Bool in
        !result.disqualified
    }

print(winner!.name) // Jessie
```

```bash
$ swiftlint lint --config .swiftlint.yml
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths
Linting 'Contents.swift' (1/1)
/Users/joeydong/Downloads/MyPlayground.playground/Contents.swift:16:14: warning: Min or Max over Sorted First or Last Violation: Prefer using `min()` or `max()` over `sorted().first` or `sorted().last` (sorted_first_last)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.38.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:
```yml
opt_in_rules:
  - sorted_first_last
```
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)?
```
Xcode 11.3
Build version 11C29
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -42,6 +42,11 @@
   [Haoocen](https://github.com/Haoocen)
   [#5023](https://github.com/realm/SwiftLint/pull/5023)
 
+* Fix false positives on `sorted_first_last` rule when `first`/`last` have
+  a predicate.
+  [woxtu](https://github.com/woxtu)
+  [#3023](https://github.com/realm/SwiftLint/issues/3023)
+
 ## 0.52.2: Crisper Clearer Pleats
 
 #### Breaking

--- a/Source/SwiftLintBuiltInRules/Rules/Performance/SortedFirstLastRule.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Performance/SortedFirstLastRule.swift
@@ -21,7 +21,11 @@ struct SortedFirstLastRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRul
             Example("myList.sorted().firstIndex(where: someFunction)"),
             Example("myList.sorted().lastIndex(where: someFunction)"),
             Example("myList.sorted().firstIndex { $0 == key }"),
-            Example("myList.sorted().lastIndex { $0 == key }")
+            Example("myList.sorted().lastIndex { $0 == key }"),
+            Example("myList.sorted().first(where: someFunction)"),
+            Example("myList.sorted().last(where: someFunction)"),
+            Example("myList.sorted().first { $0 == key }"),
+            Example("myList.sorted().last { $0 == key }")
         ],
         triggeringExamples: [
             Example("↓myList.sorted().first\n"),
@@ -50,6 +54,7 @@ private extension SortedFirstLastRule {
         override func visitPost(_ node: MemberAccessExprSyntax) {
             guard
                 node.name.text == "first" || node.name.text == "last",
+                node.parent?.is(FunctionCallExprSyntax.self) != true,
                 let firstBase = node.base?.asFunctionCall,
                 let firstBaseCalledExpression = firstBase.calledExpression.as(MemberAccessExprSyntax.self),
                 firstBaseCalledExpression.name.text == "sorted",

